external/sql/storm-sql-runtime/src/jvm/org/apache/storm/sql/runtime/ISqlTridentDataSource.java

```diff
@@ -17,14 +17,32 @@
  */
 package org.apache.storm.sql.runtime;
 
-import org.apache.storm.trident.operation.Function;
-import org.apache.storm.trident.spout.IBatchSpout;
 import org.apache.storm.trident.spout.ITridentDataSource;
+import org.apache.storm.trident.state.StateFactory;
+import org.apache.storm.trident.state.StateUpdater;
 
 /**
  * A ISqlTridentDataSource specifies how an external data source produces and consumes data.
  */
 public interface ISqlTridentDataSource {
+  class SqlTridentConsumer {
+    private StateFactory stateFactory;
+    private StateUpdater stateUpdater;
+
+    public SqlTridentConsumer(StateFactory stateFactory, StateUpdater stateUpdater) {
+      this.stateFactory = stateFactory;
+      this.stateUpdater = stateUpdater;
+    }
+
+    public StateFactory getStateFactory() {
+      return stateFactory;
+    }
+
+    public StateUpdater getStateUpdater() {
+      return stateUpdater;
+    }
+  }
+
   ITridentDataSource getProducer();
```

Comment:
how about add some javadoc for interface methods in `ISqlTridentDataSource` and `DataSource`?